test/stub-test.js

```diff
@@ -115,6 +115,15 @@ describe("stub", function () {
         assert.equals(stub(2), 0);
     });
 
+    it("should work with combination of withArgs arguments", function () {
+        var stub = createStub();
+
+        stub.withArgs(1).returns(42);
+        stub(1);
+
+        assert.isObject(stub.withArgs(1).firstCall);
```

Comment:
I would assert it's not `null` here. While being an object has the same effect, is not null is a bit more explicit.